### ISSUE TITLE
Internal Covariate Shift example in BatchNorm

### ISSUE THREAD

**Issue by vklyukin** [lab-ml/nn#23]
Hello!
The example of Internal Covariate Shift in the introduction to Batch Normalization seemed strange to me. It says that
> The paper defines Internal Covariate Shift as the change in the distribution of network activations due to the change in network parameters during training.

Although the example is:

> For example, let’s say there are two layers l1 and l2. During the beginning of the training l1 outputs (inputs to l2) could be in distribution N(0.5,1). Then, after some training steps, it could move to N(0.5,1). This is internal covariate shift.

There is no difference in parameters of the l1 outputs' distribution. Maybe there should be another values of mean and variance?

**Comment by vpj** [comment]
Sorry I didn't understand this statement

> There is no difference in parameters of the l1 outputs' distribution. Maybe there should be another values of mean and variance?

Could you please elaborate? Do you mean to say distribution of L1 outputs don't change? What do you mean by "parameters of the l1 outputs' distribution"

**Comment by vklyukin** [comment]
> Do you mean to say distribution of L1 outputs don't change?

Yes, you are right. I mean that distributions before and after the phrase "Then, after some training steps..." are the same, although the definition of Internal Covariate Shift says that it should change.

> What do you mean by "parameters of the l1 outputs' distribution"

By parameters of the distribution I mean `mean` and `variance`

**Comment by vpj** [comment]
No the distribution can change, because the parameters of the layer change. Why do you think the distribution will stay constant?

**Comment by vklyukin** [comment]
I mean, that they should change, according to the definition of Internal Covariate Shift. And phrase:
> Then, after some training steps, it could move to...

means, that the parameters of the distribution should change somehow. But it said that they moved to the `N(0.5, 1)` which is the same as it was before some training steps (the same mean and variance).

I think, that values of mean/variance after "some training steps" supposed to be differrent from the start ones, because now it's a bit confusing. For instance `N(1.5, 1)`.

**Comment by vpj** [comment]
Oh crap, really sorry I didn't notice that the example has a mistake. It should be something like N(0.5, 1) changing to N(0.7, 1.1).  Will correct it and close the issue.

Thanks again for spotting it.

**Comment by vpj** [comment]
Fixed the example: https://github.com/lab-ml/nn/commit/2e545435e8c9f3c3c831a3d45e6a5a99a42b9f28
